### PR TITLE
fix: allow group by same expr in Aggregate

### PR DESCRIPTION
**Pull request by jackwener**
# Which issue does this PR close?

Closes #.

# Rationale for this change

# What changes are included in this PR?

# Are these changes tested?

slt: groupby same colomn

# Are there any user-facing changes?

### CODE REVIEW

File: datafusion/common/src/dfschema.rs [hunk 1]
```diff
@@ -219,17 +218,7 @@ impl DFSchema {
             .map(|(idx, _)| idx);
         match matches.next() {
```
Comment (alamb) [inline datafusion/common/src/dfschema.rs:219]: Maybe this could just be `Ok(matches.next())`

**Review by alamb — APPROVED** [review]
The tests pass and the code looks good to me 👍

File: datafusion/core/tests/sqllogictests/test_files/groupby.slt [hunk 2]
```diff
@@ -58,3 +58,9 @@ SELECT 38 FROM tab0 AS cor0 GROUP BY cor0.col1, cor0.col1;
 ----
 38
 38
+
+query I rowsort
+SELECT cor0.col1 AS col2 FROM tab0 AS cor0 GROUP BY cor0.col1, cor0.col1;
```
Comment (waynexia) [inline datafusion/core/tests/sqllogictests/test_files/groupby.slt:63]: :+1:

File: datafusion/common/src/dfschema.rs [hunk 3]
```diff
@@ -217,20 +216,7 @@ impl DFSchema {
...
-                Some(_) => Err(DataFusionError::Internal(format!(
-                    "Ambiguous reference to qualified field named {}.{}",
-                    qualifier
-                        .map(|q| q.to_quoted_string())
-                        .unwrap_or("<unqualified>".to_string()),
-                    quote_identifier(name)
-                ))),
-            },
-        }
+        Ok(matches.next())
```
Comment (waynexia) [inline datafusion/common/src/dfschema.rs:219]: nit: this function won't return error now

**Review by waynexia — APPROVED** [review]
The collected exprs are for existence checking, thus duplication looks good to me. Thanks for this

File: datafusion/common/src/dfschema.rs (hunk 3 above)
Comment (jackwener) [reply datafusion/common/src/dfschema.rs:219]: it's used widely, I will try to enhance it in following PR. Thank you.

**Pull request merged**